### PR TITLE
ast, plan, expression: support builtin aggregation function 'bit_and'

### PR DESCRIPTION
**Pull request by spongedu**
support builtin aggregation function 'bit_and'

### CODE REVIEW

**Comment by mccxj** [comment]
bit_xor,bit_and,bit_or are similar, maybe we should avoid duplicate codes.

File: expression/aggregation/bit_and.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+package aggregation
+
+import (
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/context"
+	"github.com/pingcap/tidb/expression"
+	"github.com/pingcap/tidb/mysql"
+	"github.com/pingcap/tidb/sessionctx/variable"
+	"github.com/pingcap/tidb/types"
+	"math"
```
Comment (lamxTyler) [inline expression/aggregation/bit_and.go]: Put this upward.

File: expression/aggregation/bit_and.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	return ft
+}
+
+// Update implements Aggregation interface.
+func (bf *bitAndFunction) Update(ctx *AggEvaluateContext, sc *variable.StatementContext, row types.Row) error {
+	if bf.mode == CompleteMode {
+		if len(bf.Args) == 0 {
+			return nil
+		} else if len(bf.Args) != 1 {
+			return errors.New("Wrong number of args for AggFuncBitXor")
```
Comment (lamxTyler) [inline expression/aggregation/bit_and.go]: `AggFuncBitAnd`?

Comment (spongedu) [reply expression/aggregation/bit_and.go]: @lamxTyler yep. Fixed

File: expression/aggregation/bit_and.go (hunk 1 above)
Comment (spongedu) [reply expression/aggregation/bit_and.go]: ok

File: expression/aggregation/bit_and.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aggregation
+
+import (
+	"math"
```
Comment (lamxTyler) [inline expression/aggregation/bit_and.go]: You can add a blank line after this so that ci will pass.

Comment (spongedu) [reply expression/aggregation/bit_and.go]: ok

**Comment by shenli** [comment]
@lamxTyler PTAL

File: store/tikv/coprocessor.go [hunk 4]
```diff
@@ -78,7 +78,7 @@ func (c *CopClient) supportExpr(exprType tipb.ExprType) bool {
 	case tipb.ExprType_Case, tipb.ExprType_If, tipb.ExprType_IfNull, tipb.ExprType_Coalesce:
 		return true
 	// aggregate functions.
-	case tipb.ExprType_Count, tipb.ExprType_First, tipb.ExprType_Max, tipb.ExprType_Min, tipb.ExprType_Sum, tipb.ExprType_Avg:
+	case tipb.ExprType_Count, tipb.ExprType_First, tipb.ExprType_Max, tipb.ExprType_Min, tipb.ExprType_Sum, tipb.ExprType_Avg, tipb.ExprType_Agg_BitAnd:
```
Comment (coocood) [inline store/tikv/coprocessor.go]: This should be added only when it is supported by `tikv` coprocessor.

Comment (spongedu) [reply store/tikv/coprocessor.go]: @coocood ok. removed.

**Pull request merged**